force-app/main/default/classes/ClassManager.cls

```diff
@@ -0,0 +1,32 @@
+public class ClassManager {
+    public static List<Class__c> beforeDeleteClass()
+    {
+        List<Class__c> cla = [SELECT id,(SELECT id FROM Students__r WHERE Sex__c='FEMALE')  FROM Class__c WHERE id IN :Trigger.old];
```

Comment:
Try to pass `Trigger.old` as an argument, so that this method can be used in the context other than trigger.